packages/ui/src/components/Navigation/Navigation.tsx

```diff
@@ -0,0 +1,207 @@
+                                )}
+                                href="/contacts"
+                                locale={locale}
+                            >
+                                {/* TODO: Use translations */}
+                                Contacts
+                            </Link>
+                        )}
+                    </div>
+
+                    <div className="flex flex-col md:flex-row md:items-center gap-12 md:gap-4">
+                        {Boolean(isSearchEnabled && onSearch) && (
+                            <a className="hidden md:flex" href="#" onClick={handleSearch}>
+                                <MagnifyingGlassIcon className="w-[20px] h-[20px] text-gray-600 hover:text-gray-800" />
+                            </a>
+                        )}
+                        <div className="flex items-start md:items-center flex-row-reverse md:flex-row bg-gray-50 md:bg-transparent p-6 md:p-0 gap-4 justify-between md:justify-start">
+                            {languages.length > 0 && (
+                                <LanguagesDropdown
+                                    hasError={hasError}
+                                    languages={languages}
+                                    locale={locale}
+                                    currentCategory={currentCategory}
+                                    currentStory={currentStory}
+                                />
+                            )}
+                            {externalSiteLink && (
+                                <a
+                                    className="label-large text-gray-600 hover:text-gray-800 flex items-center shrink-0"
+                                    href={externalSiteLink}
+                                    target="_blank"
+                                    rel="noopener noreferrer"
+                                    data-title={
+                                        <>
+                                            {extractDomainFromUrl(externalSiteLink)}
+                                            <ArrowUpRightIcon className="ml-1 w-2 h-2" />
+                                        </>
+                                    }
+                                >
+                                    {extractDomainFromUrl(externalSiteLink)}
+                                    <ArrowUpRightIcon className="ml-1 w-2 h-2" />
+                                </a>
+                            )}
+                        </div>
+                    </div>
+                </div>
+                <div className="flex items-center gap-4 md:hidden">
+                    {Boolean(isSearchEnabled && onSearch) && (
+                        <Button
+                            className="p-0"
+                            variation="navigation"
+                            icon={MagnifyingGlassIcon}
+                            iconClassName="w-6 h-6"
```

Comment:
This feels incorrect. 
You're applying the same `className` and `iconClassName` to all navigation buttons. These styles should be part of the `navigation` variant and attached in the `Button` component, otherwise we're missing the whole point of ready-made UI components.

`iconClassName` should not be exposed from the `Button` component at all IMO.